### ISSUE TITLE
NSS log_evidence_err uses sqrt(ESS)/n_live, contradicting the same module's documented sqrt(H/n_live); it grows with run length

### ISSUE THREAD

**Issue by suchethac** [suchethac/tengri#2443]
## Summary

`log_evidence_err` is computed as `sqrt(ESS) / n_live`, which **increases** as the run
gathers more effective samples. The same module's docstring, 139 lines above, states the
correct relation. The one test that covers the value asserts only that it is positive.

`src/tengri/inference/backends/evidence.py:201` (module docstring):

> **Preset rationale**: Evidence scatter scales as σ_logZ ≈ √(H/n_live) where
> H is the entropy. "fast" trades ~2–3× wall time for σ_logZ ≈ 0.3–0.45 nats,
> fine for BMA when model differences Δlog Z ≳ 1. "accurate" provides σ_logZ
> ≈ 0.15–0.2 nats, suitable for tight model selection.

`src/tengri/inference/backends/evidence.py:340` (implementation):

```python
"log_evidence_err": float(jnp.sqrt(jnp.maximum(ess_val, 1.0)) / n_live),
```

`√(H/n_live)` and `√(ESS)/n_live` are not the same quantity and do not scale the same
way. H is a property of the problem -- how much the posterior has contracted from the
prior -- and does not grow with run length. ESS does. So a longer run reports a larger
evidence uncertainty, which is backwards for a statistical error.

## It disagrees with the module's own calibration

On a real fit (galaxy 79, configuration III, "fast" preset, n_live=100, 130 iterations,
2600 dead points) the run reported ESS 533, so the formula gives

    sqrt(533) / 100 = 0.231 nats

while the docstring's measured scatter for the "fast" preset is 0.3–0.45 nats. The
printed number is smaller than the preset's own calibrated scatter, i.e. it
under-reports, and it would keep shrinking relative to the truth on shorter runs and
growing past it on longer ones.

## Why it was not caught

`tests/.../test_evidence_route_parity.py`, `TestEvidenceDiagnostics`:

```python
def test_nss_ref_has_error(self, nss_ref):
    """NSS should report log_evidence_err in diagnostics."""
    assert "log_evidence_err" in nss_ref.diagnostics
    assert nss_ref.diagnostics["log_evidence_err"] > 0
```

`sqrt(ESS)/n_live` satisfies `> 0` for every possible input, as would any other
expression of the same ingredients. The test pins presence and sign, not meaning.

## Suggested fix, computable from what the run already holds

No extra sampling is needed. The information (entropy) is available from the dead points
the run has already produced:

    H = Σ_i w_i · log L_i  −  log Z

with `w_i` the normalized posterior weights (the same `log_weights(...)` already used by
`nested/utils.py::ess`), `log L_i` the dead points' log-likelihoods, and `log Z` the
evidence the run returns. Then

    log_evidence_err = sqrt(max(H, 0) / n_live)

which is the relation the module docstring already claims, is a property of the problem
rather than of run length, and reduces the 79/III number to something that can be
checked against the preset's calibrated 0.3–0.45 nats.

## Suggested test, to replace an assertion that cannot fail

Two properties the current test cannot see, both cheap:

1. **Direction.** On the same model and preset, a run with more live points must report a
   *smaller* `log_evidence_err`, not a larger one. `√(H/n_live)` satisfies this;
   `√(ESS)/n_live` does not.
2. **Calibration.** On a fixture with an analytic evidence, the reported error should
   bracket the actual `log Z` deviation across seeds at roughly the stated rate, or at
   minimum land within the preset's documented σ_logZ band rather than an order away.

## Impact

`log Z` itself is unaffected -- this is only the reported uncertainty on it. But the
uncertainty is what decides whether two models are distinguishable, so it matters
wherever evidence is compared: the preset docstring's own framing ("fine for BMA when
model differences Δlog Z ≳ 1") is a statement about this number. Anything doing model
selection or Bayesian model averaging on tengri evidences is reading it.

Found while comparing NSS against NUTS on a CANDELS cell. Nothing in the paper quotes
`log_evidence_err` and nothing will until this is settled.

Found at `8abb657de`.
